lib/fauna/page.rb

```diff
@@ -221,6 +221,38 @@ def reverse_each
       end
     end
 
+    ##
+    # Returns the flattened contents of the set as an array.
+    #
+    # Ideal for when you need the full contents of a set with a known small size. If you need to iterate over a set
+    # of large or unknown size, it is recommended to use +each+ instead.
+    #
+    # The set is paged in the +after+ direction.
+    def set_contents
+      each.flat_map { |x| x }
+    end
+
+    ##
+    # Iterates over the entire set, applying the configured map and discarding the result.
+    #
+    # Ideal for performing a +foreach+ over an entire set (like deleting all instances in a set). The set is iterated in
+    # the +after+ direction.
+    #
+    # Example of deleting every instance in a set:
+    #
+    #   page.apply_map! { |page_q| foreach(page_q) { |ref| delete ref } }
+    def apply_map!(&block)
```

Comment:
I made `with_map` require a collection function at the time as I wasn't sure what were all the functions that could operate with a paginate function as the collection. If only `map` and `foreach` can do that, then I think splitting it up like that is best. I agree that requiring the user to configure the expr themselves is error prone, and additionally adds more cruft to worry about.

I'm unsure how I feel about for `foreach` having to return an object you call `run!` on. I'd rather just do `with_foreach!`, though it would mean you have to call that method last if you are using builders.